### DISCUSSION TITLE
Habitat Project Repository

### DISCUSSION THREAD

**Discussion by BioNCphilic** [earthlab-education/Earth-Analytics-AY24#638] [Q&A] [answered]
I have created a [repository](https://github.com/BioNCphilic/rivercane) for the final habitat project, but the packages we've been using are not pre-loaded so I think I will need to install them in the directory I'm using. My questions are: 1. should I have put my repository on [my branch](https://github.com/earthlab-education/Earth-Analytics-AY24/tree/BioNCphilic-contrib) of the course page or did I do that part correctly? 2. If I created the repository correctly, do I load all the packages into the same folder as my directory on the computer and finally 3. Should my notebooks show up under the earth-analytics-python-env as shown below or should it be completely independent?

![Screenshot 2024-11-30 155706](https://github.com/user-attachments/assets/2101cb1f-eaa9-4b65-9509-1f3eb9bf3247)

**Comment by hanried** [comment]
I think we're supposed to create the repo for the habitat project in our own github, not in our branch of the class repo. Not sure how to answer your other questions but hopefully that helps with #1!

**Reply by BioNCphilic** [reply]
Thanks. That's what I thought too.

**Comment by eculler** [comment, marked as answer]
@BioNCphilic -- you will need to use the earth-analytics-python environment. On your computer, that should be a matter of selecting the kernel. However, I think I recall you are using Codespaces?

If you're using Codespaces, then you will need to add a file called `.devcontainer.json` to your repository. The contents of the file can match the same file in any of your assignment repositories. For example, you could copy the contents of https://raw.githubusercontent.com/earthlab-education/fundamentals-04-redlining-template/refs/heads/main/.devcontainer.json and paste them into your `.devcontainer.json` file.

Finally, if you're using Codespaces, there's a couple of considerations. 

Number 1 is your storage limit. Make sure that you have stopped and deleted any other Codespaces you are no longer using. You probably can only get away with having 2 running with this project. Make sure also that you aren't keeping a lot of data files around that you don't need -- if you download something by mistake make sure to delete it.

Number 2 is your CPU time allocation, which is about 120 hours (but the smallest machine is 2 CPUs, so that's 60 hours max). You may need to fork this repository into the `earthlab-education` account and work on it from there if you run out of free minutes. We can show you how to update the repository in your account once you're done.

**Comment by BioNCphilic** [comment]
Thanks - I was trying not to use codespaces.
I'll check it before we meet at 5:30.

On Mon, Dec 2, 2024 at 2:00 PM Elsa Culler ***@***.***> wrote:

> @BioNCphilic <https://github.com/BioNCphilic> -- you will need to use the
> earth-analytics-python environment. On your computer, that should be a
> matter of selecting the kernel. However, I think I recall you are using
> Codespaces?
>
> If you're using Codespaces, then you will need to add a file called
> .devcontainer.json to your repository. The contents of the file can match
> the same file in any of your assignment repositories. For example, you
> could copy the contents of
> https://raw.githubusercontent.com/earthlab-education/fundamentals-04-redlining-template/refs/heads/main/.devcontainer.json
> and paste them into your .devcontainer.json file.
>
> Finally, if you're using Codespaces, there's a couple of considerations.
>
> Number 1 is your storage limit. Make sure that you have stopped and
> deleted any other Codespaces you are no longer using. You probably can only
> get away with having 2 running with this project. Make sure also that you
> aren't keeping a lot of data files around that you don't need -- if you
> download something by mistake make sure to delete it.
>
> Number 2 is your CPU time allocation, which is about 120 hours (but the
> smallest machine is 2 CPUs, so that's 60 hours max). You may need to fork
> this repository into the earthlab-education account and work on it from
> there if you run out of free minutes. We can show you how to update the
> repository in your account once you're done.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/earthlab-education/Earth-Analytics-AY24/discussions/638#discussioncomment-11440212>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARFSD563TPZ3K3TCPAHXTJD2DSU3BAVCNFSM6AAAAABSY4ZYO6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBUGAZDCMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <earthlab-education/Earth-Analytics-AY24/repo-discussions/638/comments/11440212
> @github.com>
>


-- 
Randi Neff, Project Coordinator
Smoky Mountain STEM Collaborative <https://www.southwesterncc.edu/STEM>
224 Balsam, SWCC
447 College Dr., Sylva, NC 28779
phone: 828.339.4357

*I acknowledge that I live, work, and recreate on the traditional lands of
the Eastern Band of the Cherokee people.*


“The material contained in this email is based upon work supported by a
National Aeronautics and Space Administration (NASA) cooperative agreement
NNX16AB87A. Any opinions, findings, conclusions or recommendations
expressed in this material are those of the author and do not necessarily
reflect the views of NASA.”

-- 
_E-mail correspondence to and from this address may be subject to the North 
Carolina Public Records Act and shall be disclosed to third parties when 
required by Chapter 132 of the North Carolina General Statutes._
